src/bootstrap/step.rs

```diff
@@ -474,6 +474,10 @@ pub fn build_rules<'a>(build: &'a Build) -> Rules {
          .dep(|s| s.name("tool-cargo"))
          .host(true)
          .run(move |s| check::cargo(build, s.stage, s.target));
+    rules.test("check-rls", "src/tools/rls")
```

Comment:
This'll need to be coupled with a change to https://github.com/rust-lang/rust/blob/81734e0e06d24cf580dd5352c64d10110e4d3b7b/src/bootstrap/mk/Makefile.in#L57-L65 to make sure we run these tests on one of the bots I think.

Also, while you're at it, mind doing a bit of cleanup for Cargo? The `cargo` path in `test("check-cargo", "cargo")` above should be `src/tools/cargo` and the `Makefile.in` entry should change as well (the submodule has moved now)